src/Akeneo/Channel/back/tests/Integration/Query/Sql/SqlFindLocalesIntegration.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Akeneo\Test\Channel\Integration\Query\Sql;
+
+use Akeneo\Channel\API\Query\FindLocales;
+use Akeneo\Channel\API\Query\Locale;
+use Akeneo\Test\Integration\Configuration;
+use Akeneo\Test\Integration\TestCase;
+
+final class SqlFindLocalesIntegration extends TestCase
+{
+    private FindLocales $sqlFindLocales;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->sqlFindLocales = $this->get(
+            'Akeneo\Channel\Infrastructure\Query\Sql\SqlFindLocales'
+        );
+    }
+
+    public function test_it_finds_a_locale_by_its_code(): void
+    {
+        $activeLocale = $this->sqlFindLocales->find('en_US');
+        $this->assertInstanceOf(Locale::class, $activeLocale);
+        $this->assertEquals('en_US', $activeLocale->getCode());
+        $this->assertEquals(true, $activeLocale->isActivated());
+
+        $unactiveLocale = $this->sqlFindLocales->find('uk_UA');
+        $this->assertInstanceOf(Locale::class, $unactiveLocale);
+        $this->assertEquals('uk_UA', $unactiveLocale->getCode());
+        $this->assertEquals(false, $unactiveLocale->isActivated());
```

Comment:
You can directly check if it's what you want no ?
```suggestion
        $activeLocale = $this->sqlFindLocales->find('en_US');
        $this->assertEquals(new Locale('en_US', true), $activeLocale);

        $inactiveLocale = $this->sqlFindLocales->find('uk_UA');
        $this->assertEquals(new Locale('uk_UA', false), $inactiveLocale);

        $unknownLocale = $this->sqlFindLocales->find('unknown');
        $this->assertNull($unknownLocale);
```